### ISSUE TITLE
TypeError: Cannot read property 'username' of undefined

### ISSUE THREAD

**Issue by CodeTahoe** [pencilblue/pencilblue#894]
I can no longer go to `****.com/admin/content/pages`

The error I get is `TypeError: Cannot read property 'username' of undefined`

Stack Trace below:
```
2015-12-09T01:07:40.162Z - error: [1] RequestHandler: An error occurred during controller execution. URL=[GET:/admin/content/pages] ROUTE={"path":"admin/content/pages","pattern":"^/admin/content/pages[/]{0,1}$","path_vars":{},"expression":{},"themes":{"pencilblue":{"GET":{"method":"GET","path":"/admin/content/pages","access_level":2,"auth_required":true,"content_type":"text/html"}}}}
TypeError: Cannot read property 'username' of undefined
    at UserService.getFormattedName (../pencil_blue/include/service/entities/user_service.js:97:24)
    at ../pencil_blue/include/service/entities/user_service.js:140:47
    at ../pencil_blue/include/dao/dao.js:271:17
    at handleCallback (../pencil_blue/node_modules/mongodb/lib/utils.js:95:12)
    at ../pencil_blue/node_modules/mongodb/lib/cursor.js:590:16
    at handleCallback (../pencil_blue/node_modules/mongodb/node_modules/mongodb-core/lib/cursor.js:234:5)
    at setCursorDeadAndNotified (../pencil_blue/node_modules/mongodb/node_modules/mongodb-core/lib/cursor.js:424:3)
    at nextFunction (../pencil_blue/node_modules/mongodb/node_modules/mongodb-core/lib/cursor.js:586:7)
    at Cursor.next (../pencil_blue/node_modules/mongodb/node_modules/mongodb-core/lib/cursor.js:614:3)
    at fetchDocs (../pencil_blue/node_modules/mongodb/lib/cursor.js:586:10)
```
In the above I removed the full file path as that's based on the system, and doesn't really matter

**Comment by CodeTahoe** [comment]
I feel like it was something I did with content as I was editing in a wisywig on a 'page' just before this started happening... Also.. you may need this info: I'm on version: "0.4.1"... and I'd really like to not have to export my mongo collections and do a fresh install..

**Comment by CodeTahoe** [comment]
I've hacked it for now, but I don't want to deviate from a codebase yall update... 
`pencil_blue/include/service/entities/user_service.js:97:24`
```
 89     /**
 90      * Takes the specified user object and formats the first and last name.
 91      * @static
 92      * @method getFormattedName
 93      * @param {Object} user The user object to extract a name for.
 94      * @return {String} The user's full name
 95      */
 96     UserService.prototype.getFormattedName = function(user) {
 97         if( user && user.username ){
 98           var name = user.username;
 99           if (user.first_name) {
100             name = user.first_name + ' ' + user.last_name;
101           }
102           return name;
103         } else {
104           return "";
105         }
106     };
```

**Comment by CodeTahoe** [comment]
I think I figure it out... some way shape or form allowed me to submit a 'page' without an author selected, or a bad author. I've noticed an 'empty' author above my name every time, I thought it was a benign bug, but maybe it's not... (the drop down population)

**Comment by CodeTahoe** [comment]
Yep, that's what it was, I pulled my hack out after fixing the bad data, and it works... There should be a reset to default of 'none' or the uid: of 1 (the initial user / admin )

**Comment by blakecallens** [comment]
This sanitization issue was fixed in 0.5.0. Let us know if you can still replicate in that version and, if so, we'll reopen.
